packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/PropertyProvider.cs

```diff
@@ -34,7 +34,7 @@ public PropertyProvider(InputModelProperty inputProperty)
 
             Type = propertyType;
             Modifiers = MethodSignatureModifiers.Public;
-            Name = inputProperty.Name.FirstCharToUpperCase();
+            Name = inputProperty.Name.ToCleanName();
```

Comment:
We should call `ToCleanName` to remove the possible characters that are not allowed to be in an identifier, such as hyphens, and others that are not recommended to be in an identifier, such as underscores.